Code/Framework/AzToolsFramework/AzToolsFramework/Archive/ArchiveComponent.cpp

```diff
@@ -141,320 +120,474 @@ namespace AzToolsFramework
-            return false;
+            std::promise<bool> p;
+            p.set_value(false);
+            return AZStd::move(p.get_future());
         }
 
-        LaunchZipExe(m_zipExePath, commandLineArgs, createArchiveCallback, AZ::Uuid::CreateNull(), dirToArchive, false);
-        return success;
-    }
+        auto FnCreateArchive = [this, archivePath, dirToArchive](std::promise<bool>&& p) -> void
+        {
+            auto archive = m_archive->OpenArchive(archivePath, {}, AZ::IO::INestedArchive::FLAGS_CREATE_NEW);
+            if (!archive)
+            {
+                AZ_Error(s_traceName, false, "Failed to create archive file '%s'", archivePath.c_str());
+                p.set_value(false);
+                return;
+            }
 
-    void ArchiveComponent::ExtractArchive(const AZStd::string& archivePath, const AZStd::string& destinationPath, AZ::Uuid taskHandle, const ArchiveResponseCallback& respCallback)
-    {
-        ArchiveResponseOutputCallback responseHandler = [respCallback](bool result, AZStd::string /*outputStr*/) { respCallback(result); };
-        ExtractArchiveOutput(archivePath, destinationPath, taskHandle, responseHandler);
-    }
+            auto foundFiles = AzFramework::FileFunc::FindFilesInPath(dirToArchive, "*", true);
+            if (!foundFiles.IsSuccess())
+            {
+                AZ_Error(s_traceName, false, "Failed to find file listing under directory '%d'", dirToArchive.c_str());
+                p.set_value(false);
+                return;
+            }
 
-    void ArchiveComponent::ExtractArchiveOutput(const AZStd::string& archivePath, const AZStd::string& destinationPath, AZ::Uuid taskHandle, const ArchiveResponseOutputCallback& respCallback)
-    {
-        AZStd::string commandLineArgs = Platform::GetExtractArchiveCommand(archivePath, destinationPath, true);
+            bool success = true;
+            AZStd::vector<char> fileBuffer;
+            const AZ::IO::Path workingPath{ dirToArchive };
 
-        if (commandLineArgs.empty())
-        {
-            // The platform-specific implementation has already thrown its own error, no need to throw another one
-            return;
-        }
+            for (const auto& fileName : foundFiles.GetValue())
+            {
+                bool thisSuccess = false;
 
-        LaunchZipExe(m_unzipExePath, commandLineArgs, respCallback, taskHandle);
-    }
+                AZ::IO::PathView relativePath = AZ::IO::PathView{ fileName }.LexicallyRelative(workingPath);
 
-    void ArchiveComponent::ExtractArchiveWithoutRoot(const AZStd::string& archivePath, const AZStd::string& destinationPath, AZ::Uuid taskHandle, const ArchiveResponseOutputCallback& respCallback)
-    {
-        AZStd::string commandLineArgs = Platform::GetExtractArchiveCommand(archivePath, destinationPath, false);
+                AZStd::string fullPath = (workingPath / relativePath).c_str();
```

Comment:
`.Native()` should be used instead of `.c_str()` in order to construct a string from a path.
Using the `.c_str()` method calls the `AZStd::string(const char*)` constructor, which requires a `strlen` to be performed over the input string.
While `.Native()` returns an `AZStd::string&&` which will just move the string.
```suggestion
                AZStd::string fullPath = (workingPath / relativePath).Native();
```